bot/persistence/database_connector.py

```diff
@@ -438,40 +437,36 @@ def add_group_offer_and_requests(self, user_id: str,
                 db_manager.execute(queries.INSERT_GROUP_REQUEST, (user_id, course, group_nr))
             db_manager.commit()
 
-    def update_group_exchange_message_id(self, user_id: str,
-                                         course: str,
-                                         message_id: str):
+    def update_group_exchange_message_id(self, user_id: int, course: str, message_id: int):
         """Updates the message id in the GroupOffer table from 'undefined' to a valid value
 
         This function is necessary because the message_id can only be retrieved after the embed is sent, which happens
         after inserting in the db, to ensure constraints are fulfilled.
 
         Args:
-            user_id (str): The user_id of the requesting user.
+            user_id (int): The id of the requesting user.
             course (str): The course that should be exchanged.
-            message_id (str): The id of the message that contains the group exchange embed.
+            message_id (int): The id of the message that contains the group exchange embed.
         """
         with DatabaseManager(self._db_file) as db_manager:
             db_manager.execute(queries.UPDATE_GROUP_MESSAGE_ID, (message_id, user_id, course))
             db_manager.commit()
 
-    def get_candidates_for_group_exchange(self, author_id,
-                                          course: str,
-                                          offered_group: int,
-                                          requested_groups: Iterable[int]):
+    def get_candidates_for_group_exchange(self, user_id: int, course: str, offered_group: int,
+                                          requested_groups: Iterable[int]) -> Optional[List[tuple]]:
         """Gets all possible candidates for a group exchange offer.
 
         Args:
-            author_id (str): The id of the author of the request.
+            user_id (int): The id of the user who created the request.
```

Comment:
Why change this? Author is far more specific.

Reply:
"Author" could be confused with the attribute `author` of a Discord message. The author of the exchange request is SAM for example.

Reply:
But the called author was taken form ctx.author. 
So it is indeed the Author of the Message(Command) and therefor also the author of the exchange request.